src/Compilers/CSharp/Test/Semantic/Utilities/ValueSetTests.cs

```diff
@@ -0,0 +1,663 @@
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        [InlineData(4)]
+        [InlineData(-1)]
+        [InlineData(-2)]
+        [InlineData(-3)]
+        [InlineData(-4)]
+        [InlineData(int.MinValue)]
+        [InlineData(int.MaxValue)]
+        public void TestLE_01(int i1)
+        {
+            IValueSet<int> values = ForInt.Related(LessThanOrEqual, i1);
+            Assert.Equal($"[{int.MinValue}..{i1}]", values.ToString());
+        }
+
+        [Fact]
+        public void TestLE_02()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                int i1 = Random.Next(int.MinValue, int.MaxValue) + 1;
+                IValueSet<int> values = ForInt.Related(LessThanOrEqual, i1);
+                Assert.Equal($"[{int.MinValue}..{i1}]", values.ToString());
+            }
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        [InlineData(4)]
+        [InlineData(-1)]
+        [InlineData(-2)]
+        [InlineData(-3)]
+        [InlineData(-4)]
+        [InlineData(int.MinValue)]
+        [InlineData(int.MaxValue)]
+        public void TestLT_01(int i1)
+        {
+            IValueSet<int> values = ForInt.Related(LessThan, i1);
+            Assert.Equal((i1 == int.MinValue) ? "" : $"[{int.MinValue}..{i1 - 1}]", values.ToString());
+        }
+
+        [Fact]
+        public void TestLT_02()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                int i1 = Random.Next(int.MinValue, int.MaxValue) + 1;
+                IValueSet<int> values = ForInt.Related(LessThan, i1);
+                Assert.Equal($"[{int.MinValue}..{i1 - 1}]", values.ToString());
+            }
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        [InlineData(4)]
+        [InlineData(-1)]
+        [InlineData(-2)]
+        [InlineData(-3)]
+        [InlineData(-4)]
+        [InlineData(int.MinValue)]
+        [InlineData(int.MaxValue)]
+        public void TestEQ_01(int i1)
+        {
+            IValueSet<int> values = ForInt.Related(Equal, i1);
+            Assert.Equal($"[{i1}..{i1}]", values.ToString());
+        }
+
+        [Fact]
+        public void TestEQ_02()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                int i1 = Random.Next(int.MinValue, int.MaxValue);
+                IValueSet<int> values = ForInt.Related(Equal, i1);
+                Assert.Equal($"[{i1}..{i1}]", values.ToString());
+            }
+        }
+
+        [Fact]
+        public void TestIntersect_01()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                int i1 = Random.Next(int.MinValue + 1, int.MaxValue);
+                int i2 = Random.Next(int.MinValue, int.MaxValue);
+                if (i1 > i2) (i1, i2) = (i2, i1);
```

Comment:
For readability, consider removing the `if (i1 > i2)` and `if (i1 == i2)` cases in each of these tests. For instance:
```
int i1 = Random.Next(int.MinValue, int.MaxValue);
int i2 = Random.Next(i1, int.MaxValue);
```

Reply:
I prefer that both be uniformly distributed.

---
In reply to: [383429718](https://github.com/dotnet/roslyn/pull/41503#discussion_r383429718) [](ancestors = 383429718)